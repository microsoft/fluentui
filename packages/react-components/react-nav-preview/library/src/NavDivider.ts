export * from './components/NavDivider/index';
