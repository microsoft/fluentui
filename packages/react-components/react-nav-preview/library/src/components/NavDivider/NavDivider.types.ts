import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type NavDividerSlots = {
  root: Slot<'div'>;
};

/**
 * NavDivider Props
 */
export type NavDividerProps = ComponentProps<NavDividerSlots> & {};

/**
 * State used in rendering NavDivider
 */
export type NavDividerState = ComponentState<NavDividerSlots>;
// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from NavDividerProps.
// & Required<Pick<NavDividerProps, 'propName'>>
