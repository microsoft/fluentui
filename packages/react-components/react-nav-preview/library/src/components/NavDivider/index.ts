export * from './NavDivider';
export * from './NavDivider.types';
export * from './useNavDivider';
export * from './useNavDividerStyles.styles';
