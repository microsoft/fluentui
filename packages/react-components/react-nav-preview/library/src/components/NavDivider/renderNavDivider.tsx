/** @jsxRuntime automatic */
/** @jsxImportSource @fluentui/react-jsx-runtime */

import { assertSlots } from '@fluentui/react-utilities';
import type { NavDividerState, NavDividerSlots } from './NavDivider.types';

/**
 * Render the final JSX of NavDivider
 */
export const renderNavDivider_unstable = (state: NavDividerState) => {
  assertSlots<NavDividerSlots>(state);

  // TODO Add additional slots in the appropriate place
  return <state.root />;
};
