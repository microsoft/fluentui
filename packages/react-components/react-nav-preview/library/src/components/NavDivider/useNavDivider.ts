import * as React from 'react';
import type { NavDividerProps, NavDividerState } from './NavDivider.types';
import { useDivider_unstable } from '@fluentui/react-divider';

/**
 * Create the state required to render NavDivider.
 *
 * The returned state can be modified with hooks such as useNavDividerStyles_unstable,
 * before being passed to renderNavDivider_unstable.
 *
 * @param props - props from this instance of NavDivider
 * @param ref - reference to root HTMLDivElement of NavDivider
 */
export const useNavDivider_unstable = (props: NavDividerProps, ref: React.Ref<HTMLElement>): NavDividerState => {
  return useDivider_unstable({ appearance: 'strong', ...props }, ref);
};
