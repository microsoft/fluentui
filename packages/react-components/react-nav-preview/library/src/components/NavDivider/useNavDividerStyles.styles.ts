import { makeStyles, mergeClasses } from '@griffel/react';
import { useDividerStyles_unstable, type DividerSlots } from '@fluentui/react-divider';
import type { SlotClassNames } from '@fluentui/react-utilities';
import type { NavDividerState } from './NavDivider.types';

export const navDividerClassNames: SlotClassNames<DividerSlots> = {
  root: 'fui-NavDivider',
  wrapper: 'fui-NavDivider__wrapper',
};

const useStyles = makeStyles({
  root: {
    flexGrow: 0,
    marginTop: '4px',
    marginBottom: '4px',
  },
});

/**
 * Apply styling to the NavDivider slots based on the state
 */
export const useNavDividerStyles_unstable = (state: NavDividerState): NavDividerState => {
  'use no memo';

  const styles = useStyles();

  state.root.className = mergeClasses(navDividerClassNames.root, styles.root, state.root.className);
  state.wrapper.className = mergeClasses(navDividerClassNames.wrapper, state.wrapper.className);

  useDividerStyles_unstable(state);
  return state;
};
