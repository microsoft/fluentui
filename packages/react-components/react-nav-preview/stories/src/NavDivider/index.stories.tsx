import { NavDivider } from '@fluentui/react-nav-preview';

import descriptionMd from './NavDividerDescription.md';
import bestPracticesMd from './NavDividerBestPractices.md';

// export { Default } from './NavDividerDefault.stories';

export default {
  title: 'Preview Components/NavDivider',
  component: NavDivider,
  parameters: {
    docs: {
      description: {
        component: [descriptionMd, bestPracticesMd].join('\n'),
      },
    },
  },
};
