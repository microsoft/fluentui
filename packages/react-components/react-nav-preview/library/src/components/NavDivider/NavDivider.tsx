import * as React from 'react';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import { useNavDivider_unstable } from './useNavDivider';
import { renderNavDivider_unstable } from './renderNavDivider';
import { useNavDividerStyles_unstable } from './useNavDividerStyles.styles';
import type { NavDividerProps } from './NavDivider.types';

/**
 * NavDivider component - TODO: add more docs
 */
export const NavDivider: ForwardRefComponent<NavDividerProps> = React.forwardRef((props, ref) => {
  const state = useNavDivider_unstable(props, ref);

  useNavDividerStyles_unstable(state);
  // TODO update types in packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts
  // https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/custom-styling.md
  // useCustomStyleHook_unstable('useNavDividerStyles_unstable')(state);
  return renderNavDivider_unstable(state);
});

NavDivider.displayName = 'NavDivider';
