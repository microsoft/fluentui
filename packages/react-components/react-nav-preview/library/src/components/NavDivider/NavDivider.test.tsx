import * as React from 'react';
import { render } from '@testing-library/react';
import { isConformant } from '../../testing/isConformant';
import { NavDivider } from './NavDivider';

describe('NavDivider', () => {
  isConformant({
    Component: NavDivider,
    displayName: 'NavDivider',
  });

  // TODO add more tests here, and create visual regression tests in /apps/vr-tests

  it('renders a default state', () => {
    const result = render(<NavDivider>Default NavDivider</NavDivider>);
    expect(result.container).toMatchSnapshot();
  });
});
