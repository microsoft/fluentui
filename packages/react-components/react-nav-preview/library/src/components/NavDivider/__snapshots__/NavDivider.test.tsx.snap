// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDivider renders a default state 1`] = `
<div>
  <div
    class="fui-NavDivider"
  >
    Default NavDivider
  </div>
</div>
`;
